src/indexable.cr

```diff
@@ -23,34 +23,32 @@ module Indexable(T)
   # This method should only be directly invoked if you are absolutely
   # sure the index is in bounds, to avoid a bounds check for a small boost
   # of performance.
-  abstract def unsafe_at(index : Int)
+  abstract def unsafe_fetch(index : Int)
 
   # Returns the element at the given *index*, if in bounds,
-  # otherwise executes the given block and returns its value.
+  # otherwise executes the given block with the index and returns its value.
   #
   # ```
-  # a = [:foo, :bar]
-  # a.at(0) { :baz } # => :foo
-  # a.at(2) { :baz } # => :baz
+  # a = [10, 20]
+  # a.fetch(0) { |index| index * 3 } # => 10
```

Comment:
This example seems unnecessarily more complicated than the original one.

Reply:
This example shows block parameter. For example, `HTTP::Params` didn't pass it at all. Do you think it's not necessarily? Documentation of `Hash#fetch` has it.

Reply:
> But it's also wrong. The block is only invoked when the index is missing.

Yes. What is wrong? You try to get nonexistent value — you're in the block with passed key. If you try to get existent — you're not in the block.

Reply:
> Documenting the block parameter is fine. But I'd maybe make a separate example for that. Keep the main example, which demonstrates the core functionality as simple and consise as possible.
> So, I'd leave that as it was before, just replace at with fetch. Maybe also :default_value instead of :baz. This makes it easier to understand.

> And a second example could demonstrate the block parameter.

OK, done, also for `Hash#fetch` docs.

Reply:
Ah, right, it wasn't wrong.

Reply:
Ah, right, it wasn't wrong.